config/database.rb

```diff
@@ -1,13 +1,35 @@
 configure do
-  # Log queries to STDOUT in development
-  if Sinatra::Application.development?
+
+  case Sinatra::Application.environment
+  # Development database settings
+  when :development
+    # Log queries to STDOUT in development
     ActiveRecord::Base.logger = Logger.new(STDOUT)
-  end
+    set :database, {
+      adapter: "sqlite3",
+      database: "db/db.sqlite3"
+    }
+  # Test database settings
+  when :test
+    set :database, {
+      adapter: "sqlite3",
+      database: "db/test.sqlite3"
+    }
+  # Production database settings, tuned for heroku
+  when :production
+    db = URI.parse(ENV['DATABASE_URL'])  # standard heroku environment variable for configuring the database
 
-  set :database, {
-    adapter: "sqlite3",
-    database: "db/db.sqlite3"
-  }
+    set :database, {
+      :adapter => db.scheme == 'postgres' ? 'postgresql' : db.scheme,
+      :host     => db.host,
+      :username => db.user,
+      :password => db.password,
+      :database => db.path[1..-1],
+      :encoding => 'utf8'
+    }
+  else
+    raise ArgumentError("Expected RACK_ENV to be: development, test or production")
```

Comment:
This is never usually an error in Rack or Rails applications so I don't really see what's gained by making it one here.

Rather we might say something like:

```
configure :development, :test do
  set :database, {
    adapter: 'sqlite3',
    database: APP_ROOT.join('db', "#{Sinatra::Application.environment}.sqlite3")
  }
end

configure :production do
  # production db config ...
end
```

This is going to be about the same behaviour as a default Rails app, where other environments could be specified but will fail because of lack of db config (if you want to use some other environment, you have to add config for it, which is normal).